typescript/cli/src/utils/chains.ts

```diff
@@ -171,3 +171,36 @@ function handleNewChain(chainNames: string[]) {
     process.exit(0);
   }
 }
+
+/**
+ * @notice Extracts chain names from a nested configuration object
+ * @param config Object to search for chain names
+ * @return Array of discovered chain names
+ */
+export function extractChainsFromObj(config: Record<string, any>): string[] {
+  const chains: string[] = [];
+
+  // Recursively search for chain/chainName fields
+  function findChainFields(obj: any) {
+    if (obj === null || typeof obj !== 'object') return;
+
+    if (Array.isArray(obj)) {
+      obj.forEach((item) => findChainFields(item));
+      return;
+    }
+
+    if ('chain' in obj) {
+      chains.push(obj.chain);
+    }
+
+    if ('chainName' in obj) {
+      chains.push(obj.chainName);
+    }
+
+    // Recursively search in all nested values
+    Object.values(obj).forEach((value) => findChainFields(value));
+  }
+
+  findChainFields(config);
+  return chains;
```

Comment:
Maybe here instead of returning a string array, it would be better to return a set in case the same chain is present multiple times in the current object, also please update the return type to use the `ChainName` type instead of a plain string